riscv/processor.cc

```diff
@@ -519,6 +519,58 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   if (proc->extension_enabled(EXT_ZCMT))
     csrmap[CSR_JVT] = jvt = std::make_shared<jvt_csr_t>(proc, CSR_JVT, 0);
 
+  if (proc->extension_enabled_const(EXT_SMCSRIND)) {
+    miselect = std::make_shared<sscsrind_sel_csr_t>(proc, CSR_MISELECT, 0);
+    mireg = std::make_shared<sscsrind_reg_csr_t>(proc, CSR_MIREG, 0, miselect);
+    mireg2 = std::make_shared<sscsrind_reg_csr_t>(proc, CSR_MIREG2, 0, miselect);
+    mireg3 = std::make_shared<sscsrind_reg_csr_t>(proc, CSR_MIREG3, 0, miselect);
+    mireg4 = std::make_shared<sscsrind_reg_csr_t>(proc, CSR_MIREG4, 0, miselect);
+    mireg5 = std::make_shared<sscsrind_reg_csr_t>(proc, CSR_MIREG5, 0, miselect);
+    mireg6 = std::make_shared<sscsrind_reg_csr_t>(proc, CSR_MIREG6, 0, miselect);
+
+    csrmap[CSR_MISELECT] = miselect;
```

Comment:
Why not keep the existing pattern of assigning to both `csrmap` and `miselect` on a single line, instead of having to list each register twice here? With two paragraphs, there's a much bigger risk of someone changing one but not the other.

Reply:
This has undergone significant revision -- will post an update.

Reply:
Changed.